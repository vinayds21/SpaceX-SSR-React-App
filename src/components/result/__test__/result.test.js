import React from 'react';
import ReactDom from 'react-dom';

import Result from '../';

import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer'

it("renders without crashing", () => {
    const div = document.createElement('div');
    ReactDom.render(<Result></Result>,div)
})

it("renders state correctly",() => {
    const {getByTestId} = render(<Result type='loading' />)
    expect(getByTestId('app-status')).toHaveTextContent('Loading...')
})

it("matches snapshot", () => {
    const tree = renderer.create(<Result type='loading'></Result>).toJSON();
    expect(tree).toMatchSnapshot();
})