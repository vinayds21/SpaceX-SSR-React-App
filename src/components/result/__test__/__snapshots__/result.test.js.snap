// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="result"
>
  <img
    alt="image"
    src="https://static.dribbble.com/users/503653/screenshots/3143656/fluid-loader.gif"
  />
  <div
    className="text"
    data-testid="app-status"
  >
    Loading...
  </div>
</div>
`;
